### PR TITLE
Improve caching in `wp_get_global_styles_svg_filters`

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/57568

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/global-styles-and-settings.php [hunk 1]
```diff
@@ -185,15 +185,11 @@ function wp_get_global_stylesheet( $types = array() ) {
...
 	// Return cached value if it can be used and exists.
 	// It's cached by theme to make sure that theme switching clears the cache.
-	$can_use_cached = (
-		( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
-		( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG ) &&
-		( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST ) &&
-		! is_admin()
-	);
-	$transient_name = 'global_styles_svg_filters_' . get_stylesheet();
+	$can_use_cached = ! WP_DEBUG && ! is_admin();
```
Comment (felixarntz) [inline src/wp-includes/global-styles-and-settings.php]: Let's add the same comment added in the other usages of this approach (e.g. https://github.com/WordPress/wordpress-develop/commit/dd33e32767d119929e769cd21abc3b00690fb14f#diff-1be8a8089d87d1b4b6e449c0967680d6f7c170d13f247180a975cbea5e54070fR88) here too. Also, like in that linked code, the `! is_admin()` here is not necessary now and should be removed.
```suggestion
	/*
	 * Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme
	 * developer's workflow.
	 *
	 * @todo Replace `WP_DEBUG` once an "in development mode" check is available in Core.
	 */
	$can_use_cached = ! WP_DEBUG;
```

File: src/wp-includes/global-styles-and-settings.php [hunk 2]
```diff
@@ -317,5 +312,6 @@ function wp_theme_has_theme_json() {
  */
 function wp_clean_theme_json_cache() {
 	wp_cache_delete( 'wp_get_global_stylesheet', 'theme_json' );
+	wp_cache_delete( 'wp_get_global_styles_svg_filters', 'theme_json' );
```
Comment (felixarntz) [inline src/wp-includes/global-styles-and-settings.php:369]: This needs to be refreshed against `trunk` to resolve a new merge conflict :)

**Review by felixarntz — COMMENTED** [review]
@spacedmonkey Almost LGTM, just one point of feedback.

File: src/wp-includes/global-styles-and-settings.php (hunk 1 above)
Comment (oandregal) [reply src/wp-includes/global-styles-and-settings.php]: The logic for `is_admin`, `script_debug`, and `rest_request` was removed from `wp_get_global_stylesheet` for the reasons I documented here https://github.com/WordPress/gutenberg/pull/45679/files#r1025439309 So I suppose we can also remove it from here as well. @ajlende would you remember any other case in which these would be necessary?

Comment (spacedmonkey) [reply src/wp-includes/global-styles-and-settings.php]: Fixed in https://github.com/WordPress/wordpress-develop/pull/3926/commits/117e55d6e362c11208fe945520d5a6d4768de47c && https://github.com/WordPress/wordpress-develop/pull/3926/commits/ae4cbeeb17ab4200a72803dea4d6b46e946fa61c

File: src/wp-includes/global-styles-and-settings.php (hunk 2 above)
Comment (spacedmonkey) [reply src/wp-includes/global-styles-and-settings.php:369]: FIxed in https://github.com/WordPress/wordpress-develop/pull/3926/commits/5448f9cbd2d07620829437346aae973de112dc33

**Comment by spacedmonkey** [comment]
Ready for review

**Comment by hellofromtonya** [comment]
Good work @spacedmonkey. The changes are using the same code patterns in https://core.trac.wordpress.org/changeset/55148 ✅ 

What about unit tests?

File: src/wp-includes/global-styles-and-settings.php (hunk 1 above)
Comment (ajlende) [reply src/wp-includes/global-styles-and-settings.php]: > @ajlende would you remember any other case in which these would be necessary?

Nope, the SVG stuff was supposed to work exactly the same as the stylesheets.

**Review by ajlende — APPROVED** [review]
This is working for me—the caching matches what is done for the stylesheets which is good 👍

**Comment by hellofromtonya** [comment]
Note: This PR will need to be tested in the WP Admin area [per the testing instructions here](https://github.com/WordPress/wordpress-develop/pull/3712#issuecomment-1405464211) to ensure there is no fatal error from not having `wp_cache_*()` functions available when `load-styles.php` runs.

**Comment by spacedmonkey** [comment]
> Good work @spacedmonkey. The changes are using the same code patterns in https://core.trac.wordpress.org/changeset/55148 white_check_mark
> 
> What about unit tests?

Added a unit test. Sadly there are no unit tests for this function currently in core :(

File: tests/phpunit/tests/theme/wpGetGlobalStylesSvgFilters.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+		parent::tear_down();
+	}
+
+	/**
+	 * Tests that switching themes recalculates the svgs.
+	 *
+	 * @covers ::wp_get_global_styles_svg_filters
+	 *
+	 * @ticket 56970
```
Comment (felixarntz) [inline tests/phpunit/tests/theme/wpGetGlobalStylesSvgFilters.php]:
```suggestion
	 * @ticket 57568
```

File: tests/phpunit/tests/theme/wpGetGlobalStylesSvgFilters.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?php
+
```
Comment (felixarntz) [inline tests/phpunit/tests/theme/wpGetGlobalStylesSvgFilters.php]: Unnecessary blank line.
```suggestion
```

**Review by felixarntz — APPROVED** [review]
@spacedmonkey LGTM, except for 2 small nit-picks below.

**Comment by felixarntz** [comment]
@spacedmonkey @hellofromtonya I tested this in WP Admin with `CONCATENATE_SCRIPTS` enabled, `load-styles.php` is working fine as expected.

I gave it a secondary check also looking through the code paths: With this change there is no risk of breaking `load-styles.php`, since the function is only called in `wp_global_styles_render_svg_filters()`, and that function is only called in 2 specific places unrelated to `load-styles.php`.

So this should be good for `commit` IMO.

**Comment by felixarntz** [comment]
Committed in https://core.trac.wordpress.org/changeset/55185. @oandregal This should be straightforward to backport to Gutenberg.

**Comment by oandregal** [comment]
> Committed in https://core.trac.wordpress.org/changeset/55185. @oandregal This should be straightforward to backport to Gutenberg.

My understanding is that we don't do this, otherwise, it should have been done in Gutenberg _first_, correct?

Also: if the change was to be backported to Gutenberg, I'd recommend the author of the core change to create the pull request in Gutenberg, otherwise we'd run into human bottlenecks fast :)

**Pull request closed without merging**